tests/unit_tests/query_handler_runner/test_mock_query_handler_runner.py

```diff
@@ -0,0 +1,467 @@
+from pathlib import PurePosixPath
+from typing import Union
+
+import pytest
+from exasol_bucketfs_utils_python.localfs_mock_bucketfs_location import LocalFSMockBucketFSLocation
+from exasol_data_science_utils_python.schema.column import Column
+from exasol_data_science_utils_python.schema.column_name import ColumnName
+from exasol_data_science_utils_python.schema.column_type import ColumnType
+from exasol_data_science_utils_python.udf_utils.testing.mock_result_set import MockResultSet
+from exasol_data_science_utils_python.udf_utils.testing.mock_sql_executor import MockSQLExecutor
+
+from exasol_advanced_analytics_framework.query_handler.context.scope_query_handler_context import \
+    ScopeQueryHandlerContext
+from exasol_advanced_analytics_framework.query_handler.context.top_level_query_handler_context import \
+    TopLevelQueryHandlerContext
+from exasol_advanced_analytics_framework.query_handler.query.select_query import SelectQueryWithColumnDefinition, \
+    SelectQuery
+from exasol_advanced_analytics_framework.query_handler.query_handler import QueryHandler
+from exasol_advanced_analytics_framework.query_handler.result import Continue, Finish
+from exasol_advanced_analytics_framework.query_result.query_result import QueryResult
+from exasol_advanced_analytics_framework.testing.mock_query_handler_runner import MockQueryHandlerRunner
+
+
+@pytest.fixture()
+def temporary_schema_name():
+    temporary_schema_name = "temp_schema_name"
+    return temporary_schema_name
+
+
+@pytest.fixture()
+def top_level_query_handler_context(tmp_path, temporary_schema_name):
+    top_level_query_handler_context = TopLevelQueryHandlerContext(
+        temporary_bucketfs_location=LocalFSMockBucketFSLocation(base_path=PurePosixPath(tmp_path) / "bucketfs"),
+        temporary_db_object_name_prefix="temp_db_object",
+        temporary_schema_name=temporary_schema_name,
+    )
+    return top_level_query_handler_context
+
+
+class TestInput:
+    pass
+
+
+class TestOutput:
+    def __init__(self, test_input: TestInput):
+        self.test_input = test_input
+
+
+class StartFinishTestQueryHandler(QueryHandler[TestInput, TestOutput]):
+    def __init__(self, parameter: TestInput, query_handler_context: ScopeQueryHandlerContext):
+        super().__init__(parameter, query_handler_context)
+        self._parameter = parameter
+
+    def start(self) -> Union[Continue, Finish[TestOutput]]:
+        return Finish[TestOutput](TestOutput(self._parameter))
+
+    def handle_query_result(self, query_result: QueryResult) -> Union[Continue, Finish[TestOutput]]:
+        pass
+
+
+def test_start_finish(top_level_query_handler_context):
+    sql_executor = MockSQLExecutor(result_sets=[])
```

Comment:
Would be nice to have a short description about what the test actually does.

Reply:
done